pkgs/build-support/bintools-wrapper/default.nix

```diff
@@ -202,6 +202,14 @@ stdenv.mkDerivation {
         basename=$(basename "$variant")
         wrap $basename ${./ld-wrapper.sh} $variant
       done
+    ''
+    # Copy the lld-* variants if available.
+    + lib.optionalString isLLVM ''
+
+       for variant in $ldPath/${targetPrefix}lld-*; do
+        basename=$(basename "$variant")
+        wrap $basename ${./ld-wrapper.sh} $variant
```

Comment:
Will the other variants actually understand the flags added by the wrapper?

Reply:
AFAIK, there's only one variant: `lld-link` which only does Windows-style arguments.
So if the wrapper passes anything that is unexpected, it will explode.

My rationale was to progressively make wrapped `lld-link` work in the future, if this is too controversial, let's drop it from this PR.

Reply:
IMO it depends on whether it's possible to use wrapped lld-link at all.  Does a very basic use of lld-link work, or does the wrapper _always_ introduce arguments it doesn't understand?  In the latter case, I don't think it makes sense to expose a wrapped version that can't actually be used.

Reply:
If I remember well, I was able to use it by disabling enough hardening stuff, but I will double check it.
But I'm pretty sure this is a Rust specific case.

Reply:
Moving it away to a future PR.